### PR TITLE
refactor: added types to SideNavFooter

### LINKED ISSUE #13599: Add TypeScript types to SideNavFooter

### PR DESCRIPTION
**Pull request by k-rajat19**
Closes #13599

### CODE REVIEW

**Comment by k-rajat19** [comment]
I have read the DCO document and I hereby sign the DCO.

**Comment by k-rajat19** [comment]
I have read the DCO document and I hereby sign the DCO.

**Comment by k-rajat19** [comment]
@tw15egan @tay1orjones  I have fixed the conflicts, this is PR ready for review

File: packages/react/src/components/UIShell/SideNavFooter.tsx [hunk 1]
```diff
@@ -12,6 +12,13 @@ import React from 'react';
 import PropTypes from 'prop-types';
 import { usePrefix } from '../../internal/usePrefix';
 
+export interface SideNavFooterProps {
+  assistiveText: string;
+  className?: string;
+  expanded: boolean;
+  onToggle: (event: React.MouseEvent<HTMLButtonElement>) => void;
```
Comment (tw15egan) [inline packages/react/src/components/UIShell/SideNavFooter.tsx:33]: Can we add descriptions for each type? These can be copied from the `PropType` definitions

Comment (k-rajat19) [reply packages/react/src/components/UIShell/SideNavFooter.tsx:33]: Done :)

**Review by tw15egan — APPROVED** [review]

**Review by andreancardona — APPROVED** [review]
thank you for contributing! 🎉

**Pull request merged**